src/System.Xml.XmlSerializer/tests/Performance/XsPerformanceTest.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.IO;
+using System.Runtime.Serialization;
+
+namespace System.Xml.XmlSerializer.Tests.Performance
+{
+    internal class XsSerializerFactory : SerializerFactory
+    {
+        private static XsSerializerFactory _instance = null;
+        public static XsSerializerFactory GetInstance()
+        {
+            if (_instance == null)
+            {
+                _instance = new XsSerializerFactory();
+            }
+            return _instance;
+        }
+
+        public override IPerfTestSerializer GetSerializer(SerializerType serializerType)
+        {
+            switch (serializerType)
+            {
+                case SerializerType.XmlSerializer:
+                    return new XsSerializer();
+                default:
+                    throw new NotImplementedException();
```

Comment:
I don't think either of them are quite appropriate as you are saying either the method is not implemented of it's not supported which isn't the case. I think either ArgumentException or ArgumentOutOfRangeException would be better as the problem is with what was requested and not that the GetSerializer method isn't functional. Either way though, this is test code so it's not super important :smile: